### PR TITLE
Update c-api-cpu.yml: change DoEsrp to true

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update c-api-cpu.yml: change the default value of DoEsrp to true

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by snnn** [comment]
We want to run the publish nuget step by default.
There are two options: 
1. Change the default value of doesrp from false to true.
2. Remove doesrp from the conditions of   https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml#L498

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ parameters:
 - name: DoEsrp
   displayName: Run code sign tasks? Must be true if you are doing an Onnx Runtime release.
   type: boolean
-  default: false
+  default: true
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:20]: Any reason to run esrp for nightly releases as well?

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:20]: Is it ok to publish an unsigned package to our 1st partners?

Comment (pranavsharma) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:20]: It doesn't hurt to publish signed pkgs as such. I asked becoz they are just nightlies, not intended for production usage.

Comment (pranavsharma) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:20]: I didn't however get how this prevented nightlies from getting published.

**Review by pranavsharma — APPROVED** [review]

**Pull request closed without merging**